### PR TITLE
Add more granular control of Kokkos pair and fix comm

### PR DESCRIPTION
**Pull request by stanmoore1**
**Summary**

Currently there is no way to run Kokkos pair and fix forward comm on the host. This PR adds more granular control of Kokkos pair and fix comm through Kokkos package options. 

**Related Issue(s)**

None.

**Author(s)**

Stan Moore (SNL)

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

This PR changes the Kokkos package `cuda/aware` keyword to `gpu/aware`, since Kokkos supports both CUDA and HIP.

### CODE REVIEW

File: src/KOKKOS/out.txt [hunk 1]
```diff
@@ -0,0 +1,44 @@
+pair_buck_coul_cut_kokkos.cpp
```
Comment (rbberger) [inline src/KOKKOS/out.txt:1]: probably these two files shouldn't be here

Comment (stanmoore1) [reply src/KOKKOS/out.txt]: Yes good catch, I removed them.

**Comment by stanmoore1** [comment]
I tested both `fix qeq` and `pair eam`, everything seems to be working.

**Comment by stanmoore1** [comment]
This PR will almost certainly collide with #2533. Once #2533 is merged, I will fix the conflicts then it should be ready.

**Comment by stanmoore1** [comment]
A few ReaxFF regression tests are failing...

**Comment by stanmoore1** [comment]
Kokkos regression tests passed, should be ready for merge.

**Comment by akohlmey** [comment]
@stanmoore1 how about accepting both, "cuda/aware" and "gpu/aware", but document only the latter as a way to preserve backward compatibility for a transition period? Some time in the future we could then remove the older keyword.

**Comment by stanmoore1** [comment]
Yes that sounds great to me.

**Comment by akohlmey** [comment]
> Yes that sounds great to me.

done

**Comment by stanmoore1** [comment]
Thank you, this looks good to me.

**Review by akohlmey — APPROVED** [review]

**Pull request merged**